okhttp/src/main/java/okhttp3/internal/tls/CertificateChainCleaner.java

```diff
@@ -17,56 +17,43 @@
 package okhttp3.internal.tls;
 
 import java.security.GeneralSecurityException;
-import java.security.PublicKey;
 import java.security.cert.Certificate;
 import java.security.cert.X509Certificate;
 import java.util.ArrayDeque;
 import java.util.ArrayList;
 import java.util.Deque;
 import java.util.Iterator;
-import java.util.LinkedHashMap;
 import java.util.List;
-import java.util.Map;
 import javax.net.ssl.SSLPeerUnverifiedException;
-import javax.security.auth.x500.X500Principal;
 
 /**
- * A set of trusted Certificate Authority (CA) certificates that are trusted to verify the TLS
- * certificates offered by remote web servers.
+ * Computes the effective certificate chain from the raw array returned by Java's built in TLS APIs.
+ * Cleaning a chain returns a list of certificates where the first element is {@code chain[0]}, each
+ * certificate is signed by the certificate that follows, and the last certificate is a trusted CA
+ * certificate.
+ *
+ * <p>Use of the chain cleaner is necessary to omit unexpected certificates that aren't relevant to
+ * the TLS handshake and to extract the trusted CA certificate for the benefit of certificate
+ * pinning.
  *
  * <p>This class includes code from <a href="https://conscrypt.org/">Conscrypt's</a> {@code
  * TrustManagerImpl} and {@code TrustedCertificateIndex}.
  */
-public final class CertificateAuthorityCouncil {
-  private final Map<X500Principal, List<X509Certificate>> subjectToCaCerts = new LinkedHashMap<>();
+public final class CertificateChainCleaner {
```

Comment:
Most of this change is splitting the data (trust root index) from behavior (cleaning certificates).